packages/front-end/components/OpenVisualEditorLink.tsx

```diff
@@ -45,49 +61,81 @@ const OpenVisualEditorLink: FC<{
     });
 
     return url;
-  }, [visualEditorUrl, id, changeIndex, apiHost]);
+  }, [apiHost, changeIndex, id, visualEditorUrl]);
 
-  const navigate = useCallback(() => {
-    track("Open visual editor", {
-      source: "visual-editor-ui",
-      status: "success",
+  const getVisualEditorKey = useCallback(async () => {
+    const res = await apiCall<{ key: string }>("/visual-editor/key", {
+      method: "GET",
     });
-    window.location.href = url;
+    return res.key;
+  }, [apiCall]);
+
+  const navigate = useCallback(async () => {
+    setShowExtensionDialog(false);
+
+    setIsNavigating(true);
+
+    const key = await getVisualEditorKey();
+
+    window.postMessage(
+      {
+        type: "GB_REQUEST_OPEN_VISUAL_EDITOR",
+        data: key,
+      },
+      window.location.origin
+    );
+
+    // in the case a user has clicked 'proceed anyway' when we do not detect the
+    // chrome extension installation, we ignore waiting for the responsem msg
+    // and navigate right away.
+    if (isBypassing) {
+      setIsNavigating(false);
+
+      track("Open visual editor", {
+        source: "visual-editor-ui",
+        status: "success",
```

Comment:
We might want to change the status here to `bypass` or something.  Ideally we would see very few people hit this code.  This would let us verify that with data and make sure the auto-auth flow is working for people as intended.

Reply:
Addressed